src/book/02-system/02-subsystems/04-motion.mdx

```diff
@@ -5,3 +5,66 @@ title: Motion
+
+A walk engine will trigger when a [**walkcommand**](https://github.com/NUbots/NUbots/blob/master/shared/message/motion/WalkCommand.proto) is [emitted](/system/nubots/nuclear#emit-statements). The walk command gives a vector [x, y, z] which represents:
+
+| Variable | Description                   |
+| -------- | ----------------------------- |
+| x        | velocity in metres per second |
+| y        | velocity in metres per second |
+| z        | angle in radians per second   |
+
+The [behaviour](system/nubots/behaviour) of the robot determines what the walk command will be.
+
+## Open Loop Hybrid Walk
+
+This walk is an **open loop hybrid walk controller**, dynamically switching between an analytically solved trajectory for the centre of mass based on the **linear inverted pendulum model**, and a **Zero Moment Point preview controller**. To add local feedback, positions for the pitch and roll ankle joints are prescribed based on information from the Inertial Measurement Unit (IMU) to counteract small imbalances.
+
+This walk is based on the work of [Yi et al.](https://doi.org/10.1109/HUMANOIDS.2013.7029960) and Song ('Development of an Omni-directional Gait Generator and a Stabilization Feedback Controller for Humanoid Robots'). The code is a refactored version of the [UPennalizers](https://fling.seas.upenn.edu/~robocup/wiki/) / Team DARwIn [2013 code release](https://github.com/UPenn-RoboCup/UPennalizers/tree/master). NUbots have made several improvements and simplifications.
+
+The module for this walk is located in the [OldWalkEngine module folder](https://github.com/NUbots/NUbots/tree/master/module/motion/OldWalkEngine).
+
+## Quintic Spline Walk
+
+The Quintic Walk is an **open loop controller** based on **quintic splines** (piecewise fifth degree polynomials). It sets points in each direction for rotation and translation and uses those points to create smooth quintic splines.
+
+The engine has support for kicking, however this is not used at this time.
+
+The code used for this walk comes from [**BitBots**](https://github.com/bit-bots/bitbots_motion/), which in turn was based off [Quentin "Leph" Rouxel and Team Rhoban's walk](https://github.com/Rhoban/model/).
+
+The implementation of this walk is currently a work in progress. See the implementation on the [QuinticWalk branch](https://github.com/NUbots/NUbots/tree/biddulph/quintic_walk/module/motion/QuinticWalk).
+
+### Implementation
+
+The walk uses the Footstep class to generate the targets for the foot. It handles the special cases of side stepping and turning to prevent collisions with the other foot.
+
+The SplineContainer class holds Points and multiple Spline objects. As the SplineContainer gets points, it creates Splines based on these points. These Spline objects are matched to some name in a mapping.
```

Comment:
How does it create splines based on these points? Is that beyond the scope of this doc?

Reply:
I think that sort of thing is beyond the scope of this. I tried to find some good external resources for splines but the only page I could find dedicated to splines was on Wikipedia... Might add some info in the maths page, but it's not something specific to NUbots.

Reply:
Is it worth adding links to the codebase for each of the classes mentioned?